feat: make GroClient() args optional

jli opened the pull request:
Changes `GroClient.__init__` so that both API host and access token are optional arguments. The API host defaults to the production API server and the access token defaults to `$GROAPI_TOKEN`. This saves clients (and us) from needing to spell out 'api.gro-intelligence.com' over and over.

Also added docstring for the constructor and added to developer docs: https://developers.gro-intelligence.com/jli-convenient-constructor/api.html#creating-a-client

Commit:
feat: make GroClient() args optional

Changes GroClient.__init__ so that both API host and access token are
optional arguments. The API host defaults to the production API server
and the access token defaults to `$GROAPI_TOKEN`. This saves clients
(and us) from needing to spell out 'api.gro-intelligence.com' over and
over.

Also added docstring for the constructor, and added to developer docs.

nemozen commented on groclient/client.py:
```diff
@@ -57,7 +58,36 @@ def __init__(self, response, retry_count, url, params):
+            >>> client = GroClient(access_token="your_token_here")
+        """
+        if access_token is None:
+            env_token = os.environ.get("GROAPI_TOKEN")
```
we don't really need a separate variable here,  just use access_token rather than setting it on line 90

jli replied:
done

nemozen reviewed (commented):
lgtm just one minor comment

Commit:
add destructor checks for clean shutdown

This changes ensures _async_http_client and _ioloop are always defined,
and adds checks that they are set before trying to close them in the
destructor. Without this, if the constructor fails early (because
access_token isn't set), the destructor raises exceptions trying to
access and call methods on the unset values.

Noticed this running pytest on Python 3.9. For whatever reason, pytest
on Python 3.6 didn't complain.

Commit:
minor cleanup: reuse access_token local variable
